src/main/java/io/supertokens/oauth/HttpRequest.java

```diff
@@ -0,0 +1,189 @@
+            }
+            return getResponse(con);
+        } catch (FileNotFoundException e) {
+            throw new OAuthClientNotFoundException();
+        }
+    }
+
+    public static Response doJsonPut(String url, Map<String, String> queryParams, Map<String, String> headers, JsonObject jsonInput) throws IOException, OAuthClientNotFoundException {
+        try {
+            if (queryParams == null) {
+                queryParams = new HashMap<>();
+            }
+            URL obj = new URL(url + "?" + queryParams.entrySet().stream()
+                    .map(e -> e.getKey() + "=" + URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8))
+                    .collect(Collectors.joining("&")));
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.setRequestMethod("PUT");
+            con.setConnectTimeout(CONNECTION_TIMEOUT);
+            con.setReadTimeout(READ_TIMEOUT);
+            con.setDoOutput(true);
+            con.setRequestProperty("Content-Type", "application/json");
+
+            if (headers != null) {
+                for (Map.Entry<String, String> entry : headers.entrySet()) {
+                    con.setRequestProperty(entry.getKey(), entry.getValue());
+                }
+            }
+
+            try (DataOutputStream os = new DataOutputStream(con.getOutputStream())) {
+                os.writeBytes(jsonInput.toString());
+            }
+            return getResponse(con);
+        } catch (FileNotFoundException e) {
+            throw new OAuthClientNotFoundException();
+        }
+    }
+
+    public static Response doJsonDelete(String url, Map<String, String> headers, JsonObject jsonInput) throws IOException, OAuthClientNotFoundException {
+        try {
+            URL obj = new URL(url);
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.setRequestMethod("DELETE");
+            con.setConnectTimeout(CONNECTION_TIMEOUT);
+            con.setReadTimeout(READ_TIMEOUT);
+            con.setDoOutput(true);
+            con.setRequestProperty("Content-Type", "application/json");
+
+            if (headers != null) {
+                for (Map.Entry<String, String> entry : headers.entrySet()) {
+                    con.setRequestProperty(entry.getKey(), entry.getValue());
+                }
+            }
+
+            if (jsonInput != null) {
+                try (DataOutputStream os = new DataOutputStream(con.getOutputStream())) {
+                    os.writeBytes(jsonInput.toString());
+                }
+            }
+
+            return getResponse(con);
+        } catch (FileNotFoundException e) {
+            throw new OAuthClientNotFoundException();
+        }
+    }
+
+    private static Response getResponse(HttpURLConnection con) throws IOException {
+        int responseCode = con.getResponseCode();
+        InputStream inputStream;
+        if (con.getErrorStream() != null) {
```

Comment:
I have now reverted the original HttpRequest to not contain any changes. Although this one is similar, it's tailored a bit to be able to capture returned status code, headers and responses from error cases. I do not want to introduce these things into the original one so that I don't accidentally introduce any issues

Reply:
In this case, would it make sense to rename the class&file itself and/or add some kind of description about what makes this different?

Reply:
done